### ISSUE TITLE
Do you have the BatchNorm layers frozen?

### ISSUE THREAD

**Issue by insmod-he** [jwyang/faster-rcnn.pytorch#170]
Hi, jwyang:

    Do you have the BatchNorm layers frozen during the training process?

**Comment by clavichord93** [comment]
I think all batch normalization layers are frozen during the training.

Please refer to the lines 261 to 267 and 278 to 284 in `faster-rcnn.pytorch/lib/model/faster_rcnn/resnet.py`.

**Comment by insmod-he** [comment]
Yes you are right, thanks for your reply.

**Comment by askerlee** [comment]
wonder why they are frozen? On a different dataset the statistics may be quite different, and batchnorm may help

**Comment by clavichord93** [comment]
@askerlee When training two-detectors, batch size is commonly reletively small, so batch normalization statistics are not accurate if not frozen. This can harm detection performance.

**Comment by askerlee** [comment]
@clavichord93 thanks :) but I think this problem can be addressed by setting a high momentum of the batchnorm layer. The running mean and running var are tracking the mean and var using exponential moving average. The default momentum 0.1 attenuates the old inputs too quickly. Setting it to something like 0.8~0.9 should make it able to track a much longer history (i.e., statistics across multiple iterations).

**Comment by clavichord93** [comment]
@askerlee First, I think you misunderstand the meaning of momentum in BN. A small momentum tends to preserve more old values. Second, the inaccurate BN statistics not only affects the inference, but the training. This can not be solved by changing momentum. The only way to resolve this is to use large batch size.

You can refer to this paper: https://arxiv.org/abs/1711.07240.

**Comment by askerlee** [comment]
@clavichord93 thanks for clarifying that. Yeah I misunderstood how the momentum is used. 

Though, using 0.9, the far-from-optimal value, I still got much faster training speed on my dataset (20% better accuracy after 10000 iterations, and 3% better when the training ends), and the trained model has less severe overfitting. I'll try to see how the model performs when using the default 0.1 momentum.

Of course, that's just the benefits observed on my own dataset. I'm not sure how (or whether) it benefits on other datasets.
